src/main.rs

```diff
@@ -57,7 +61,7 @@ async fn main() -> Result<(), Box<dyn Error>> {
     let host = IpAddr::from_str(&host_raw)?;
     let port = env::var("PORT").unwrap_or_else(|_| "80".into()).parse()?;
 
-    let client = Client::new(env::var("DISCORD_TOKEN")?);
+    let clients: ClientMap = Arc::new(DashMap::new());
```

Comment:
This implementation will lead to bloat over time because clients are never removed. This needs an expiry on it or needs to be made into an LRU.